test/integration/controller/core/workload_controller_test.go

```diff
@@ -426,4 +427,38 @@ var _ = ginkgo.Describe("Workload controller", ginkgo.Ordered, ginkgo.ContinueOn
 			gomega.Expect(finalQueueWorkload.Spec.Priority).To(gomega.Equal(&initialPriority))
 		})
 	})
+
+	ginkgo.When("the workload has a maximum execution time set", func() {
+		ginkgo.It("should deactivate the workload when the time expires", func() {
+			// due time rounding in conditions, the workload will stay admitted
+			// for a time between maxExecutionTime - 1s and maxExecutionTime
+			maxExecTime := 2 * time.Second
+			wl := testing.MakeWorkload("wl", ns.Name).
+				Queue("lq").
+				MaximumExecutionTimeSeconds(int32(maxExecTime.Seconds())).
+				Obj()
+			key := client.ObjectKeyFromObject(wl)
+			ginkgo.By("creating the workload and reserving its quota", func() {
+				gomega.Expect(k8sClient.Create(ctx, wl)).To(gomega.Succeed())
+				admission := testing.MakeAdmission("cq").Obj()
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, key, wl)).To(gomega.Succeed())
+					g.Expect(util.SetQuotaReservation(ctx, k8sClient, wl, admission)).To(gomega.Succeed())
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+			ginkgo.By("waiting for the workload to be admitted", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, key, wl)).To(gomega.Succeed())
+					g.Expect(workload.IsAdmitted(wl)).To(gomega.BeTrue())
+					g.Expect(workload.IsActive(wl)).To(gomega.BeTrue())
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+			ginkgo.By("waiting for the workload to be deactivated", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, key, wl)).To(gomega.Succeed())
+					g.Expect(workload.IsActive(wl)).To(gomega.BeFalse())
```

Comment:
Additionally, could you check if the workload has an appropriate Deativated condition?

Reply:
it will leave room for flakynes, that condition is removed after the wl gets deactivated, and as long as the workload is no longer active we can safely say "job done"

Reply:
Kindly ping > @trasc

Reply:
Added condition check, (for metrics it requires creating the queues ... , the code path for metrics increment is already tested and I don't see too much added value )

Reply:
>  (for metrics it requires creating the queues ... , the code path for metrics increment is already tested and I don't see too much added value )

That makes sense. Thanks.